cmd/launcher/svc_config_windows.go

```diff
@@ -103,6 +122,62 @@ func checkDependOnService(launcherServiceKey registry.Key, logger log.Logger) {
 	// Set service to depend on Dnscache
 	serviceList = append(serviceList, dnscacheService)
 	if err := launcherServiceKey.SetStringsValue(dependOnServiceName, serviceList); err != nil {
-		level.Error(logger).Log("msg", "could not set strings value for DependOnService", "err", err)
+		logger.Log(context.TODO(), slog.LevelError,
+			"could not set strings value for DependOnService",
+			"err", err,
+		)
+	}
+}
+
+// checkRestartActions checks the current value of our `SERVICE_FAILURE_ACTIONS_FLAG` and
+// sets it to true if required. See https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-service_failure_actions_flag
+func checkRestartActions(logger *slog.Logger) {
```

Comment:
this method is broadly named because there are more changes coming, just wanted to keep this PR small

Reply:
Is the follow-up PR going to configure the failure actions? Or are they already configured elsewhere?

Reply:
they are already configured via Wix,  but yes - this will be re-setting those with a lower restart count timeout value that wix doesn't expose